sway-core/src/asm_generation/from_ir.rs

```diff
@@ -346,6 +367,77 @@ impl<'ir> AsmBuilder<'ir> {
         });
     }
 
+    // Read the argument(s) base from the script data.
+    fn read_args_value_from_script_data(&mut self, function: &Function, reg: &VirtualRegister) {
+        // Check if enough data was sent.
+        let data_len_reg = self.reg_seqr.next();
+        self.bytecode.push(Op {
+            opcode: either::Either::Left(VirtualOp::GTF(
+                data_len_reg.clone(),
+                VirtualRegister::Constant(ConstantRegister::Zero),
+                VirtualImmediate12 {
+                    value: GTFArgs::ScriptDataLength as u16,
+                },
+            )),
+            comment: "Get length of script data".into(),
```

Comment:
Are main args the only script data we have or are the SDKs allowed to have more things in there (such as the content of a `Vec` in a contract's ABI?)

Reply:
Yep. That's the reason the check isn't an EQ but a GTE. In fuels-ts's Multicall we use this for sending some variable length data:

https://github.com/FuelLabs/fuels-ts/blob/5d4d6ce7fa1a23deae3f41be94c9fe2ee9851772/packages/contract/src/contracts/multicall/src/main.sw#L53-L56